loader/src/main/java/net/neoforged/fml/loading/ClasspathTransformerDiscoverer.java

```diff
@@ -59,4 +70,38 @@ private void locateTransformers(String resource) throws IOException {
             found.add(new NamedPath(path.toUri().toString(), path));
         }
     }
+
+    private void scanModClasses() {
+        final Map<String, List<Path>> modClassPaths = getModClasses();
+        modClassPaths.forEach((modid, paths) -> {
+            if (shouldLoadInServiceLayer(paths)) {
+                found.add(new NamedPath(modid, paths.toArray(Path[]::new)));
```

Comment:
If a jar is loaded in the service layer like this, its modid should be blacklisted from being loaded by the normal mod locator. (similar to `ModDirTransformerDiscoverer#allExcluded()`)

Reply:
Right